packages/access-token/src/access-token.js

```diff
@@ -45,15 +54,27 @@ export async function getAccessToken(
   // Ensure we're protected with a unique accessToken for this basedir.
   await fs.promises.mkdir(sharedStateDir, { mode: 0o700, recursive: true });
 
-  // Ensure an access token exists.
-  const { storage, commit, close } = openJSONStore(sharedStateDir);
-  const accessTokenKey = `accessToken/${port}`;
-  if (!storage.has(accessTokenKey)) {
-    storage.set(accessTokenKey, await generateAccessToken());
-    await commit();
+  let accessToken;
+
+  const lockFilePath = path.join(sharedStateDir, DATA_FILE);
+  const release = await lockfile.lock(sharedStateDir, {
+    lockFilePath,
```

Comment:
This feels a little leaky. Can we instead add a function returned by `openJSONStore` to `acquireLock` / "start a transaction" ? Then the lock can be automatically released if taken when `close`-ing the store.

Reply:
The problem is that `openJSONStore` itself must take the lock, or else the `.has` and `.get` can be out of sync with what's on disk.  I could add `openJSONStore(dir, exclusive = false)` where `exclusive = true` would do the semantics I need and not be a leaky abstraction.

Reply:
Okay, I see away to do `acquireLock` now.